lib/src/components/Alert/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { Children, cloneElement, forwardRef } from 'react'
+
+import { Button } from '@/components/Button'
+import type { ButtonProps } from '@/components/Button/types'
+import { Text } from '@/components/Text'
+import { VariantIcon } from '@/components/VariantIcon'
+import { classNames } from '@/utils'
+import { CloseButton } from '@old/CloseButton'
+
+import alertStyles from './alert.module.scss'
+import type { AlertProps, AlertTitleProps, CloneActionsReturns } from './types'
+
+const cx = classNames(alertStyles)
+
+const AlertComponent = forwardRef<HTMLDivElement, AlertProps>(
+  (
+    {
+      children,
+      className,
+      cta,
+      dataTestId,
+      handleClose,
+      hideIcon,
+      isFullWidth,
+      size = 'sm',
+      variant,
+      ...rest
+    },
+    ref
+  ) => {
+    const closeButtonDataTestId = dataTestId ? `${dataTestId}-close-button` : undefined
+    const defaultVariantIcon = variant === 'beige' ? undefined : variant
+    const withAiButton = variant === 'ai'
+    const showIcon = !hideIcon
+
+    const content = Children.toArray(children).map((child: React.ReactElement) => {
+      if (child.type === AlertTitle)
+        return cloneElement(child, { hasCloseButton: !!handleClose, variant: size })
+
+      return child
+    })
+
+    // Handle clone actions recursively in case of multiple buttons
+    const cloneActions = (child: React.ReactElement): CloneActionsReturns => {
+      if (child) {
+        if (child.type === AlertButton) {
+          // If Alert variant is ai, we override the CTA Buttons to use the AI sub-variants
+          return cloneElement<ButtonProps>(child, {
+            size,
+            variant: withAiButton ? 'primary-ai' : undefined,
+          })
+        }
+        if (child.type === AlertSecondaryButton) {
+          return cloneElement<ButtonProps>(child, {
+            size,
+            variant: withAiButton ? 'tertiary-ai' : undefined,
+          })
+        }
+
+        if (child.props?.children) {
+          return cloneElement(child, {
+            ...child.props,
+            children: Children.map(child.props.children, (nestedChild: React.ReactElement) =>
+              cloneActions(nestedChild)
+            ),
+          })
+        }
+      }
+
+      return child
+    }
+
+    const actions = React.isValidElement(cta) ? cloneActions(cta) : cta
```

Comment:
Do we really need all this ? Or can we just make use of `Alert.AlertButton` etc. For the `size` it would be simple in CSS to set / override the size of child buttons etc without cloning elements.